Framework/PythonInterface/mantid/utils/nomad/diagnostics.py

```diff
@@ -0,0 +1,450 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+# 3rd-party imports
+import numpy
+import numpy as np
+import yaml
+
+# standard imports
+from collections import namedtuple
+import enum
+import random
+import string
+from typing import List, NamedTuple
+
+
+class CollimationLevel(enum.Enum):
+    r"""Collimation state of an eight-pack"""
+    Empty = 0
+    Half = 1
+    Full = 2
+
+
+class InstrumentComponentLevel(enum.IntEnum):
+    """
+    Instrument component level
+    """
+    Panel = 0  # a.k.a "Panel"
+    EightPack = 1
+    Tube = 2
+    Pixel = 3
+
+
+class _NOMADMedianDetectorTest:
+    r"""
+    Mixin providing methods to algorithm NOMADMedianDetectorTest
+
+    The following attributes should've been defined and initialized in NOMADMedianDetectorTest:
+        intensities: numpy.ma.core.MaskedArray
+        config: dict
+    """
+
+    # Instrument geomtry
+    # TODO these quantities should be derived from the instrument object
+    MONITOR_COUNT = 2  # number of upstream monitors
+    PANEL_COUNT = 6
+    PANEL_FLAT = [False, False, False, False, True, True]  # last two panels are flat
+    EIGHTPACK_COUNT = 99
+    TUBES_IN_EIGHTPACK = 8
+    TUBE_COUNT = EIGHTPACK_COUNT * TUBES_IN_EIGHTPACK
+    PIXELS_IN_TUBE = 128
+    PIXELS_IN_EIGHTPACK = TUBES_IN_EIGHTPACK * PIXELS_IN_TUBE
+    PIXEL_COUNT = TUBE_COUNT * PIXELS_IN_TUBE
```

Comment:
A lot of content within this file are repeated from other files, what is the reason behind this code duplication at this extreme level?